config.org
#+TITLE: EMACS CONFIG

* CONFIG

** OPTIONS

*** Scroll bar, Tool bar, Menu bar

#+BEGIN_SRC emacs-lisp
(tool-bar-mode -1)
(scroll-bar-mode -1)
;; (menu-bar-mode -1)
#+END_SRC

*** Lockfiles
#+BEGIN_SRC emacs-lisp
(setq create-lcokfiles nil)
#+END_SRC

*** Backup
#+BEGIN_SRC emacs-lisp
(setq backup-directory-alist `(("." . "~/.saves")))
(setq backup-by-copying t)
#+END_SRC


** PACKAGES

*** try

#+BEGIN_SRC emacs-lisp
(use-package try
	:ensure t)
#+END_SRC

*** which-key

#+BEGIN_SRC emacs-lisp
(use-package which-key
	:ensure t 
	:config
	(which-key-mode))
#+END_SRC
*** dashbpoard

#+BEGIN_SRC emacs-lisp
(use-package dashboard
	:config
	(dashboard-setup-startup-hook))
#+END_SRC
*** org
**** org-agenda
#+BEGIN_SRC emacs-lisp
(global-set-key "\C-ca" 'org-agenda)
(setq org-agenda-files (list "~/Documents/GitHub/themathguy.github.io/"))
(setq org-todo-keywords
       '((sequence "TODO" "NEXT" "WAITING" "INACTIVE" "|" "DONE" "CANCELLED")))
#+END_SRC
**** org-srco
#+BEGIN_SRC emacs-lisp
(setq org-src-fontify-natively t)
#+END_SRC
*** evil

#+BEGIN_SRC emacs-lisp
(use-package evil
	:ensure t
	:config
	(evil-mode))
#+END_SRC

*** company

#+BEGIN_SRC emacs-lisp
(use-package company
	:ensure t
	:config
	(company-mode))
(add-hook 'after-init-hook 'global-company-mode)	
(setq company-idle-delay t)
#+END_SRC

*** helm

#+BEGIN_SRC emacs-lisp
(use-package helm
	:ensure t)
(global-set-key (kbd "M-x") #'helm-M-x)
(global-set-key (kbd "C-x r b") #'helm-filtered-bookmarks)
(global-set-key (kbd "C-x C-f") #'helm-find-files)
(helm-autoresize-mode t)
#+END_SRC

*** projectile

#+BEGIN_SRC emacs-lisp
(use-package projectile
	:ensure t
	:config
	(projectile-mode))
#+END_SRC

*** helm-projectile

#+BEGIN_SRC emacs-lisp
(use-package helm-projectile
	:ensure t
	:config
	(helm-projectile-on))
#+END_SRC

*** solaire

#+BEGIN_SRC emacs-lisp
(use-package solaire-mode
	:ensure t
	:config
	(solaire-mode))
;; brighten buffers (that represent real files)
(add-hook 'after-change-major-mode-hook #'turn-on-solaire-mode)
;; To enable solaire-mode unconditionally for certain modes:
(add-hook 'ediff-prepare-buffer-hook #'solaire-mode)

;; ...if you use auto-revert-mode:
(add-hook 'after-revert-hook #'turn-on-solaire-mode)

;; highlight the minibuffer when it is activated:
(add-hook 'minibuffer-setup-hook #'solaire-mode-in-minibuffer)

#+END_SRC




** THEME

*** Window Size

#+BEGIN_SRC emacs-lisp
(defun set-frame-size-according-to-resolution ()
  (interactive)
  (if window-system
  (progn
    ;; use 120 char wide window for largeish displays
    ;; and smaller 80 column windows for smaller displays
    ;; pick whatever numbers make sense for you
    (if (> (x-display-pixel-width) 1280)
           (add-to-list 'default-frame-alist (cons 'width 120))
           (add-to-list 'default-frame-alist (cons 'width 80)))
    ;; for the height, subtract a couple hundred pixels
    ;; from the screen height (for panels, menubars and
    ;; whatnot), then divide by the height of a char to
    ;; get the height we want
    (add-to-list 'default-frame-alist 
         (cons 'height (/ (- (x-display-pixel-height) 0)
                             (frame-char-height)))))))

(set-frame-size-according-to-resolution)
#+END_SRC

*** Transparency

#+BEGIN_SRC emacs-lisp
  (set-frame-parameter (selected-frame) 'alpha 80)
#+END_SRC

#+BEGIN_SRC emacs-lisp
 (defun transparency (value)
   "Sets the transparency of the frame window. 0=transparent/100=opaque"
   (interactive "nTransparency Value 0 - 100 opaque:")
   (set-frame-parameter (selected-frame) 'alpha value))
#+END_SRC

*** Theme

#+BEGIN_SRC emacs-lisp
(use-package doom-themes
	:ensure t)
(load-theme 'doom-one t)
(doom-themes-org-config)
(doom-themes-visual-bell-config)
(setq frame-title-format "emacs")
#+END_SRC





newold/config.org
#+TITLE: EMACS CONFIGURATION

* Emacs Config

** Startup options

*** Scroll bar, Tool bar, Menu bar

#+BEGIN_SRC emacs-lisp

(tool-bar-mode -1)
(scroll-bar-mode -1)
;; (menu-bar-mode -1)

#+END_SRC


*** Cursor Type

#+BEGIN_SRC emacs-lisp
(set-default 'cursor-type 'hbar)
#+END_SRC




*** Org SRC
#+BEGIN_SRC emacs-lisp
(setq org-src-fontify-natively t)
#+END_SRC



*** ENVIRONMENT


** Utilities

*** Ido Mode

#+BEGIN_SRC emacs-lisp

(ido-mode)

#+END_SRC

*** Column Number Mode

#+BEGIN_SRC emacs-lisp
(column-number-mode)
#+END_SRC

*** Show-Paren Mode

#+BEGIN_SRC emacs-lisp
(show-paren-mode)
(setq show-paren-delay 0)
(require 'paren)
(set-face-background 'show-paren-match "VioletRed4")
#+END_SRC

*** Hl Line Mode

#+BEGIN_SRC emacs-lisp
(global-hl-line-mode 1)
(set-face-foreground 'highlight nil)
(set-face-background hl-line-face "gray13")
(set-face-attribute hl-line-face nil :underline nil)
#+END_SRC


*** Winner Mode
#+BEGIN_SRC emacs-lisp
(winner-mode t)
#+END_SRC



    
*** Calc


#+BEGIN_SRC emacs-lisp
(org-babel-do-load-languages
   'org-babel-load-languages
      '((calc . t)))
#+END_SRC

** Keybindings

*** Wind Move

Lets you move point from window to window using Shift and arrow keys. 

#+BEGIN_SRC emacs-lisp
(windmove-default-keybindings)
#+END_SRC


*** TODO Which Key 

** Package-initialize

*** Ace Window

#+BEGIN_SRC emacs-lisp
(global-set-key (kbd "M-p") 'ace-window)
#+END_SRC

*** Ace Jump Mode

#+BEGIN_SRC emacs-lisp
(add-to-list 'load-path "/full/path/where/ace-jump-mode.el/in/")
(autoload
  'ace-jump-mode
  "ace-jump-mode"
  "Emacs quick move minor mode"
  t)
;; you can select the key you prefer to
(define-key global-map (kbd "C-c SPC") 'ace-jump-mode)
#+END_SRC

*** Evil Mode
*** Org Mode

*** Neotree


#+BEGIN_SRC emacs-lisp
(add-to-list 'load-path "/some/path/neotree")
(require 'neotree)
(global-set-key [f8] 'neotree-toggle)
(setq neo-theme (if (display-graphic-p) 'ascii 'arrow))
(setq-default neo-show-hidden-files t)
#+END_SRC


*** Initialize

#+BEGIN_SRC emacs-lisp
(package-initialize)
#+END_SRC

** Theme



#+BEGIN_SRC emacs-lisp

(load-theme 'misterioso)

(setq frame-title-format "emacs")

#+END_SRC


config.el
(tool-bar-mode -1)
(scroll-bar-mode -1)
;; (menu-bar-mode -1)

(setq create-lcokfiles nil)

(setq backup-directory-alist `(("." . "~/.saves")))
(setq backup-by-copying t)

(use-package try
	:ensure t)

(use-package which-key
	:ensure t 
	:config
	(which-key-mode))

(use-package dashboard
	:config
	(dashboard-setup-startup-hook))

(global-set-key "\C-ca" 'org-agenda)
(setq org-agenda-files (list "~/Documents/GitHub/themathguy.github.io/"))
(setq org-todo-keywords
       '((sequence "TODO" "NEXT" "WAITING" "INACTIVE" "|" "DONE" "CANCELLED")))

(setq org-src-fontify-natively t)

(use-package evil
	:ensure t
	:config
	(evil-mode))

(use-package company
	:ensure t
	:config
	(company-mode))
(add-hook 'after-init-hook 'global-company-mode)	
(setq company-idle-delay t)

(use-package helm
	:ensure t)
(global-set-key (kbd "M-x") #'helm-M-x)
(global-set-key (kbd "C-x r b") #'helm-filtered-bookmarks)
(global-set-key (kbd "C-x C-f") #'helm-find-files)
(helm-autoresize-mode t)

(use-package projectile
	:ensure t
	:config
	(projectile-mode))

(use-package helm-projectile
	:ensure t
	:config
	(helm-projectile-on))

(use-package solaire-mode
	:ensure t
	:config
	(solaire-mode))
;; brighten buffers (that represent real files)
(add-hook 'after-change-major-mode-hook #'turn-on-solaire-mode)
;; To enable solaire-mode unconditionally for certain modes:
(add-hook 'ediff-prepare-buffer-hook #'solaire-mode)

;; ...if you use auto-revert-mode:
(add-hook 'after-revert-hook #'turn-on-solaire-mode)

;; highlight the minibuffer when it is activated:
(add-hook 'minibuffer-setup-hook #'solaire-mode-in-minibuffer)

(defun set-frame-size-according-to-resolution ()
  (interactive)
  (if window-system
  (progn
    ;; use 120 char wide window for largeish displays
    ;; and smaller 80 column windows for smaller displays
    ;; pick whatever numbers make sense for you
    (if (> (x-display-pixel-width) 1280)
           (add-to-list 'default-frame-alist (cons 'width 120))
           (add-to-list 'default-frame-alist (cons 'width 80)))
    ;; for the height, subtract a couple hundred pixels
    ;; from the screen height (for panels, menubars and
    ;; whatnot), then divide by the height of a char to
    ;; get the height we want
    (add-to-list 'default-frame-alist 
         (cons 'height (/ (- (x-display-pixel-height) 0)
                             (frame-char-height)))))))

(set-frame-size-according-to-resolution)

(set-frame-parameter (selected-frame) 'alpha 80)

(defun transparency (value)
  "Sets the transparency of the frame window. 0=transparent/100=opaque"
  (interactive "nTransparency Value 0 - 100 opaque:")
  (set-frame-parameter (selected-frame) 'alpha value))

(use-package doom-themes
	:ensure t)
(load-theme 'doom-one t)
(doom-themes-org-config)
(doom-themes-visual-bell-config)
(setq frame-title-format "emacs")
